consensus/mint-client/src/bin/main.rs

```diff
@@ -10,119 +10,158 @@ use mc_consensus_api::{
-                amount,
-                view_public_key: *recipient.view_public_key(),
-                spend_public_key: *recipient.spend_public_key(),
-                nonce,
-                tombstone_block,
+            let tx = params
+                .try_into_mint_tx(|| {
+                    let last_block_info = blockchain_api
+                        .get_last_block_info(&Empty::new())
+                        .expect("get last block info");
+                    last_block_info.index + MAX_TOMBSTONE_BLOCKS - 1
+                })
+                .expect("failed creating tx");
+            let resp = client_api
+                .propose_mint_tx(&(&tx).into())
+                .expect("propose tx");
+            println!("response: {:?}", resp);
+        }
+
+        Commands::GenerateMintTx { out, params } => {
+            let tx = params
+                .try_into_mint_tx(|| panic!("missing tombstone block"))
+                .expect("failed creating tx");
+
+            let json = to_string_pretty(&tx).expect("failed serializing tx");
+
+            fs::write(out, json).expect("failed writing output file");
+        }
+
+        Commands::SubmitMintTx { node, tx_filenames } => {
+            // Load all txs.
+            let txs: Vec<MintTx> = load_json_files(&tx_filenames);
+
+            // All tx prefixes should be the same.
+            if !txs.windows(2).all(|pair| pair[0].prefix == pair[1].prefix) {
+                panic!("All txs must have the same prefix");
+            }
+
+            // Collect all signatures.
+            let mut signatures = txs
+                .iter()
+                .flat_map(|tx| tx.signature.signatures())
+                .cloned()
+                .collect::<Vec<_>>();
+            signatures.sort();
+            signatures.dedup();
+
+            let merged_tx = MintTx {
+                prefix: txs[0].prefix.clone(),
+                signature: MultiSig::new(signatures),
             };
 
-            let message = prefix.hash();
-            let signature = MultiSig::new(vec![signer.try_sign(message.as_ref()).unwrap()]);
-            let tx = MintTx { prefix, signature };
+            let env = Arc::new(EnvBuilder::new().name_prefix("mint-client-grpc").build());
+            let ch = ChannelBuilder::default_channel_builder(env).connect_to_uri(&node, &logger);
+            let client_api = ConsensusClientApiClient::new(ch);
 
             let resp = client_api
-                .propose_mint_tx(&(&tx).into())
+                .propose_mint_tx(&(&merged_tx).into())
                 .expect("propose tx");
             println!("response: {:?}", resp);
         }
     }
 }
+
+fn load_json_files<T: DeserializeOwned>(filenames: &[PathBuf]) -> Vec<T> {
+    filenames
+        .iter()
+        .map(|filename| {
+            let bytes = fs::read_to_string(filename)
+                .unwrap_or_else(|err| panic!("Failed reading file {:?}: {}", filename, err));
+            serde_json::from_str(&bytes)
```

Comment:
Nit: These are not bytes.
```suggestion
            let json = fs::read_to_string(filename)
                .unwrap_or_else(|err| panic!("Failed reading file {:?}: {}", filename, err));
            serde_json::from_str(&json)
```